homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -55,7 +55,9 @@ async def generate_content(call: ServiceCall) -> ServiceResponse:
             if not Path(image_filename).exists():
                 raise HomeAssistantError(f"`{image_filename}` does not exist")
             mime_type, _ = mimetypes.guess_type(image_filename)
-            if mime_type is None or not mime_type.startswith("image"):
+            if mime_type is None or (
+                not mime_type.startswith("image") and mime_type != "application/pdf"
```

Comment:
The parameter of the action is called image_filename. We need to add another one and deprecate the existing one.

Reply:
In that case, should it have any MIME type restriction at all?